cpp/tests/io/parquet_test.cpp

```diff
@@ -192,6 +194,31 @@ std::unique_ptr<cudf::column> make_parquet_list_col(
                offsets_size, offsets.release(), std::move(child), 0, rmm::device_buffer{});
 }
 
+// function to read parquet file metadata from the footer.
```

Comment:
This comment should tell us what the return value means. I couldn't figure out why it returned an `int`. It's an exit status of some kind.

Because it's a check for validity (?) and not _returning_ the footer, a name like `validate_footer` instead of `read_footer` would be better.

I'm trying to figure out exactly what it is verifying, which is testing the limits of my prior knowledge of Parquet. The existence of a footer? The validity of the footer's contents? Both? Can a footer be empty and still valid? Comments explaining what each failure (`return -1`) means could help.

Reply:
Also it seems this only returns `0` or `-1`. Would a `bool` return be clearer?

Reply:
The purpose of the function is to populate the passed-in FileMetaData object, which is then used to find the first data page.  Maybe `get_footer then`?  The validation isn't the point, but is necessary before continuing.  I'm not clear on what the proper thing to do in a test is when something unexpected occurs before the test can even be performed.  Should this just throw an exception if the footer is invalid?  The only EXPECT that is really part of the test is the final `EXPECT_EQ(ph.data_page_header.num_values, page_rows);` Suggestions?

Reply:
@bdice I fixed up the documentation for read_footer and took your suggestion to return bool.  

Looking at other tests, should I instead be using CUDF_EXPECTS when doing validity checking not strictly part of what's being tested?